Model/Filter/FilterBuilder.php

```
@@ -0,0 +1,207 @@
+                break;
+            case 'select':
+            case 'multiselect':
+                $value = $item->getData('label');
+                break;
+            case 'date':
+                break;
+            case 'boolean':
+                $value = (bool) $item->getData('value');
+                break;
+            default:
+                $this->logger->debug(sprintf(
+                    'Cannot build include filter for "%s" frontend input type',
+                    $frontendInput
+                ));
+                return;
+        }
+        $this->mapValueToFilter($filterName, $value);
+    }
+
+    /**
+     * @param string $name
+     * @param string|array $value
+     */
+    private function mapValueToFilter(string $name, $value)
+    {
+        if ($this->brand === $name) {
+            $this->includeFilters->setBrands($this->makeArrayFromValue($value));
+            return;
+        }
+
+        switch (strtolower($name)) {
+            case 'price':
+                $this->includeFilters->setPrice(min($value), max($value));
+                break;
+            case 'new':
+                $this->includeFilters->setFresh($value);
+                break;
+            default:
+                $this->includeFilters->setCustomFields($name, $this->makeArrayFromValue($value));
+                break;
+        }
+    }
+
+    /**
+     * @return IncludeFilters
+     */
+    public function getIncludeFilters()
+    {
+        return $this->includeFilters;
+    }
+
+    /**
+     * @return ExcludeFilters
+     */
+    public function getExcludeFilters()
+    {
+        return $this->excludeFilters;
+    }
+
+    /**
+     * @param string|int|array $value
+     * @return array
+     */
+    private function makeArrayFromValue($value)
+    {
+        if (is_string($value)) {
+            $value = [$value];
+        }
+        if (is_numeric($value)) {
+            $value = [$value];
+        }
+        return $value;
```

Why two if statements for the same action? Use OR conditional to simplify this.

---

Should we throw an exception if the value is not string, numeric or array?